src/t_hash.c

```diff
@@ -989,7 +989,7 @@ void hrandfieldWithCountCommand(client *c, long l, int withvalues) {
     int uniq = 1;
     robj *hash;
 
-    if ((hash = lookupKeyReadOrReply(c,c->argv[1],shared.null[c->resp]))
+    if ((hash = lookupKeyReadOrReply(c,c->argv[1],shared.emptyarray))
```

Comment:
seems like a copy paste bug.
modeled after SRANDMEMBER, which indeed returns a `shared.emptyarray` in that case.

i see this is documented for both [sets](https://redis.io/commands/srandmember), and [hashes](https://redis.io/commands/hrandfield) / zsets

Reply:
Yes. I found this problem in the process of using HRANDFIELD. So i also check for others, and found the same problem in ZRANDMEMBER

Reply:
yes, ZRAND* and HRAND* were added together and modeled after SRAND*, so it's no surprise they both have the same bugs.